pkg/controller/controller.go

```diff
@@ -392,10 +392,26 @@ func (c *RolloutController) hasStatefulSetNotReadyPods(sts *v1.StatefulSet) (boo
 	// a terminating pod (which we don't consider "ready") may have not yet failed the
 	// readiness probe for the consecutive number of times required to switch its status
 	// to not-ready. For this reason, we list all StatefulSet pods and check them one-by-one.
-	notReadyPods, err := c.listNotReadyPodsByStatefulSet(sts)
+	//
+	// First we check that we see at least the number of pods desired by the Replicas field.
+	// Sometimes it takes a while until pods are created after terminating them.
+	// We consider the missing pods as not ready.
+	pods, err := c.listPodsByStatefulSet(sts)
 	if err != nil {
 		return false, err
 	}
+
+	if len(pods) < int(sts.Status.Replicas) {
+		level.Info(c.logger).Log(
+			"msg", "StatefulSet status is reporting all pods ready, but the rollout operator found less pods than expected",
```

Comment:
> StatefulSet status is reporting all pods ready

I don't think this is true. We haven't checked their readiness yet. I would update the log message to better reflect the check we've done.

Reply:
The message refers to the fact that if we're here, then `sts.Status.Replicas == sts.Status.ReadyReplicas`.

Reply:
Yes, but doesn't mean "all pods ready". It's just a wrong message we're sending through the logs.

Reply:
Sorry, I'm wrong. All good here.